include/aspect/plugins.h

```diff
@@ -392,6 +390,11 @@ namespace aspect
                      const std::string &documentation,
                      ParameterHandler  &prm)
       {
+        AssertThrow (name != "null",
+                     ExcMessage(std::string("A parameter required by this model has not "
+                                            "been specified by this parameter file. "
+                                            "The required parameter reports: <")
+                                + documentation + ">"));
```

Comment:
This should never happen in practice. I think all the model classes which call this function have their own asserts which should be hit before this gets called (ie geometry_model/interface.cc throws "You need to select a Geometry model ('set Model name' in 'subsection Geometry model')." which is a somewhat more clear message than this (This would throw: "A plugin required by this model has not been specified by this parameter file. Additional information from the required plugin: <Geometry model::model name>").

I think this is a good backup, but maybe it should be an Assert, rather than an AssertThrow, because it should only exist to warn developers that they need their own, more useful, AssertThrow. Thoughts?